lxd/request/request.go

```diff
@@ -78,7 +78,8 @@ func CountStartedRequest(r *http.Request) {
 // The callback function is used to mark the request as completed for the API metrics.
 func MetricsCallback(request *http.Request, result metrics.RequestResult) {
 	callback, err := GetCtxValue[func(metrics.RequestResult)](request.Context(), MetricsCallbackFunc)
-	if err != nil && (strings.HasPrefix(request.URL.Path, "/1.0") || request.URL.Path == "/") {
+	// If host is a custom socket the request is coming from the /dev/lxd socket.
+	if err != nil && strings.HasPrefix(request.URL.Path, "/1.0") && request.Host != "custom.socket" {
 		// Log a warning if endpoint is part of the main API, and therefore should be counted fot the API metrics.
 		logger.Warn("Request will not be counted for the API metrics", logger.Ctx{"url": request.URL.Path, "method": request.Method, "remote": request.RemoteAddr})
```

Comment:
This feels like debug logging to me. Maybe it's not even needed as you suggested @hamistao.
That said, it seems to have brought the question of why `/dev/lxd` requests ended up in that function.

Reply:
I think a warning maybe makes more sense since a sys admin may want to know that some requests are not being counted as expected as this should help understand inconsistencies in the metrics values. That said I am fine with removing it if we are unable to properly check for devlxd requests.

Reply:
The warning is incorrect in that it is logging for a request that it shouldnt be tracking.
So we need to fix the logic that causes this to be logged on every request, but we can leave the general idea of logging a warning if it occurs for another unexpected reason.